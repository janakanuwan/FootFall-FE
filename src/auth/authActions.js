import { createAction } from 'redux-actions';
import { FETCH_USER, LOGOUT_USER, SET_USER } from '../const/action-types';

const logoutUser = createAction(
  LOGOUT_USER,
  user => ({ user }),
);

const setUser = createAction(
  SET_USER,
  userResponse => userResponse,
);

const fetchUser = createAction(
  FETCH_USER,
  loginInfo => loginInfo,
);

export { logoutUser, fetchUser, setUser };
