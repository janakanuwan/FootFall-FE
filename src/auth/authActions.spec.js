import { fetchUser, logoutUser, setUser } from './authActions';

describe('userActions', () => {
  it('should create an action for user logout', () => {
    expect(logoutUser({ name: 'Hello', email: 'hello@gamil.com' })).toMatchSnapshot();
  });

  it('should create an action for set user', () => {
    expect(setUser({
      token: '1233',
      user: { email: 'hello@test.com', password: '1234', rememberMe: true }
    })).toMatchSnapshot();
  });

  it('should create an action for set failed user ', () => {
    const action = setUser(new Error('Invalid Email'));
    expect(action).toMatchSnapshot();
    expect(action.error).toEqual(true);
    expect(action.payload.message).not.toEqual(null);

  });

  it('should create an action for user login request', () => {
    expect(fetchUser({ email: 'hello@test.com', password: '1234', rememberMe: true })).toMatchSnapshot();
  });
});

