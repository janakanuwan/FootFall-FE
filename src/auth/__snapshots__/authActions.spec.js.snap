// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userActions should create an action for set failed user  1`] = `
Object {
  "error": true,
  "payload": [Error: Invalid Email],
  "type": "SET_USER",
}
`;

exports[`userActions should create an action for set user 1`] = `
Object {
  "payload": Object {
    "token": "1233",
    "user": Object {
      "email": "hello@test.com",
      "password": "1234",
      "rememberMe": true,
    },
  },
  "type": "SET_USER",
}
`;

exports[`userActions should create an action for user login request 1`] = `[Function]`;

exports[`userActions should create an action for user logout 1`] = `
Object {
  "payload": Object {
    "user": Object {
      "email": "hello@gamil.com",
      "name": "Hello",
    },
  },
  "type": "LOGOUT_USER",
}
`;
